### PR TITLE
[semantic-arc-opts] Perform a rename/add some comments after talking with @atrick.

### PR DESCRIPTION
**Pull request by gottesmm**
This should add additional clarity to the pass's impl.

### CODE REVIEW

**Comment by gottesmm** [comment]
@swift-ci smoke test

File: lib/SILOptimizer/Transforms/SemanticARCOpts.cpp [hunk 1]
```diff
@@ -71,9 +89,18 @@ class LiveRange {
...
+  /// If we are able to eliminate this LiveRange due to it being from a
+  /// guaranteed value, we must flip the ownership of all of these instructions
+  /// to guaranteed from owned.
+  ///
+  /// NOTE: Normally only destroying or consuming uses end the live range. We
+  /// copy these transitive uses as well into the consumingUses array since
+  /// transitive uses can extend a live range up to an unreachable block without
+  /// ultimately being consuming. In such a situation if we did not also store
+  /// this into consuming uses, we would not be able to ascertain just using the
+  /// "consumingUses" array the true lifetime of the LiveRange.
```
Comment (atrick) [inline lib/SILOptimizer/Transforms/SemanticARCOpts.cpp:103]: s/LiveRange/OwnershipLiveRange

**Review by atrick — APPROVED** [review]
Thanks

File: lib/SILOptimizer/Transforms/SemanticARCOpts.cpp (hunk 1 above)
Comment (gottesmm) [reply lib/SILOptimizer/Transforms/SemanticARCOpts.cpp]: Fixed.

**Comment by gottesmm** [comment]
@swift-ci smoke test

**Comment by gottesmm** [comment]
@swift-ci smoke test

**Comment by gottesmm** [comment]
@swift-ci smoke test

**Comment by gottesmm** [comment]
@swift-ci smoke test

**Comment by gottesmm** [comment]
@swift-ci test windows platform

**Comment by gottesmm** [comment]
@swift-ci smoke test

**Pull request merged**